development_bank_wales/pipeline/predictive_model/plotting.py

```diff
@@ -0,0 +1,325 @@
+    )
+
+    plt.tight_layout()
+
+    # Move plot to the right
+    x_fig.subplots_adjust(right=RIGHT)
+
+    # Set color bar
+    cbar_ax = x_fig.add_axes([0.605, 0.15, 0.02, 0.7])
+    cbar = x_fig.colorbar(im_0, cax=cbar_ax)
+    cbar.ax.tick_params(labelsize=24)
+
+    # plt.title(title)
+    plt.savefig(FIG_PATH / (title.replace(":", " -") + ".png"), format="png", dpi=500)
+
+    # Show
+    plt.show()
+
+
+def plot_confusion_matrix(
+    solutions, predictions, label_set=None, title="", plot_type="plt"
+):
+    """Plot the confusion matrix for different classes given correct labels and predictions.
+
+    Args:
+        solutions (np.array) : Ground truth labels.
+        predictions (np.array): Predicted labels.
+        label_set (list, optional): Set list of labels. Defaults to None.
+        title (str, optional): Plot title. Defaults to "".
+        plot_type (str, optional): Whether to use "plt" or "searborn" style. Defaults to "plt".
+    """
+
+    if plot_type == "plt":
+
+        # Compute confusion matrix
+        cm = sklearn.metrics.confusion_matrix(
+            solutions, predictions, labels=range(len(label_set))
+        )
+
+        # Set figure size
+        if len(label_set) > 5:
+            plt.figure(figsize=(10, 10))
+        else:
+            plt.figure(figsize=(3, 3))
+
+        # Plot  confusion matrix with blue color map
+        plt.imshow(cm, interpolation="none", cmap="Blues")
+
+        # Write out the number of instances per cell
+        for (i, j), z in np.ndenumerate(cm):
+            plt.text(j, i, z, ha="center", va="center")
+
+        # Assign labels and title
+        plt.xlabel("Prediction")
+        plt.ylabel("Ground truth")
+        title = "Confusion Matrix for {}".format(title)
+        plt.title(title)
+
+        # Set x ticks and labels
+        plt.gca().set_xticks(range(len(label_set)))
+        plt.gca().set_xticklabels(label_set, rotation=50)
+
+        # Set y ticks and labels
+        plt.gca().set_yticks(range(len(label_set)))
+        plt.gca().set_yticklabels(label_set)
+        plt.gca().invert_yaxis()
+
+        plt.savefig(
+            FIG_PATH / (title.replace(":\n", " -") + ".png"),
+            format="png",
+            dpi=500,
+            bbox_inches="tight",
+        )
+
+        # Show plot
+        plt.show()
+
+    else:
+
+        matrix = confusion_matrix(solutions, predictions)
+        matrix = matrix.astype("float") / matrix.sum(axis=1)[:, np.newaxis]
+
+        # Build the plot
+        plt.figure(figsize=(10, 7))
+        sns.set(font_scale=1.4)
+        sns.heatmap(
+            matrix,
+            annot=True,
+            annot_kws={"size": 10},
+            cmap=plt.cm.Greens,
+            linewidths=0.2,
+        )
+
+        # Add labels to the plot
+        tick_marks = np.arange(len(label_set))
+        tick_marks2 = tick_marks + 0.5
+        plt.xticks(tick_marks, label_set, rotation=25)
+        plt.yticks(tick_marks2, label_set, rotation=0)
```

Comment:
To improve readability:
```suggestion
        tick_marks_x = np.arange(len(label_set))
        tick_marks_y = tick_marks_x + 0.5
        plt.xticks(tick_marks_x, label_set, rotation=25)
        plt.yticks(tick_marks_y, label_set, rotation=0)
```